### ISSUE TITLE
Any plans to make new release?

### ISSUE THREAD

**Issue by kloczek** [qtwebkit/qtwebkit#1053]
Last alpha4 was released year ago and since than have been added +45k commits.

**Comment by Biswa96** [comment]
Is this project dead?

**Comment by kloczek** [comment]
Looks like last commit was at Oct 1, 2020.

**Comment by GNUuser** [comment]
@annulen ??

**Comment by onkobu** [comment]
- being phased out for Gentoo Linux: https://packages.gentoo.org/packages/dev-qt/qtwebkit
- takes Openshot into the abyss: https://packages.gentoo.org/packages/media-video/openshot, https://github.com/OpenShot/openshot-qt/issues/4320
- Blog dead: http://qtwebkit.blogspot.com/ (2017)
- Maling list dead: https://lists.webkit.org/pipermail/webkit-qt/ (May 2021, possibly the wrong list)
- IRC is empty (and the freenode-link changed)

**Comment by RJVB** [comment]
knock knock?

**Comment by annulen** [comment]
@RJVB Given that I'm IP blocked from download.qt.io, I don't see any possibility to continue work on this project. Maybe @mnutt can take it over, though.

**Comment by kloczek** [comment]
Just humble Q: do you really need to have an access to download.qt.io to be able make new release? 🤔

**Comment by RJVB** [comment]
On Sunday May 22 2022 05:25:15 Konstantin Tokarev wrote:
***@***.*** Given that I'm IP blocked from download.qt.io, I don't see any possibility to continue work on this project. Maybe @mnutt can take it over, though.

IP blocked?? You been naughty or just living in the wrong country at the moment?

Does this mean that there is a newer version on qt.io than what's currently on github? I've only been following the developments here (others too, it appears)...!

**Comment by annulen** [comment]
@RJVB More likely, on the wrong planet.

@kloczek Nope, just don't want to deal with people who treat me like crap

**Comment by mitya57** [comment]
There are official mirrors. I just tried a few, and they work from this part of planet: https://download.qt.io/static/mirrorlist/.
For example:
- https://www.nic.funet.fi/pub/mirrors/download.qt-project.org/
- https://ftp.jaist.ac.jp/pub/qtproject/
- https://ftp2.nluug.nl/languages/qt/
- https://mirrors.ocf.berkeley.edu/qt/

**Comment by RJVB** [comment]
***@***.*** More likely, on the wrong planet.

Lol, that must be an advantage if you're working on world wide web stuff! ;)

**Comment by annulen** [comment]
Oh, how nice, great white masters have left a back door for lowly untermenschen. I guess I should be awestruck by that (no).

**Comment by onkobu** [comment]
I also thought a new release requires some interaction with this Git repo – I am completely unaware of the true nature and complexity of Qt component releases. In addition – also didn't try this and I am certain illegal measures against internet safety are illegal – some _untermenschen_ reported successful tunneling through onion based internet technology. (From my experience coding and/or release management are totally unrelated to prejudices. And I am in doubt that the need for a release is a platform for political theatre and personal insults.)

**Comment by annulen** [comment]
@onkobu I fully agree with you. Anyone who is currently using Qt is welcome to step in and do release management and/or coding for this project. Release management part is thoroughly documented in wiki, but feel free to ask if you have any questions

**Comment by RJVB** [comment]
Wouldn't it already be a big step forward if the WebKit part would be updated, touching as little of the Qt wrapper/glue code as possible?

I always thought WebKit was an independent (from the Qt Company, since Qt5.x anyway) project, with IIRC at least 4 front-ends that are part of the project (Apple/ObjC, GTk, Qt, Enlightenment). That would also mean its main repo is hosted elsewhere. If the Qt Comp want to keep a hand in the Qt front-end that's undoubtedly going to be Qt6 based (which isn't going to interest me much for the foreseeable future) ... so the Qt5 version could remain a community effort.

I'd offer to help with that but given my time, lack of relevant experience and sufficiently powerful hardware I can't do much more than beta-testing.

**Comment by annulen** [comment]
@RJVB: See https://github.com/qtwebkit/qtwebkit/wiki/Overview-of-(Qt)WebKit-projects. In short, you're mostly correct. WebKit is independent project, its main repo is currently https://github.com/WebKit/WebKit (qtwebkit/qtwebkit used to be a direct fork of upstream WebKit, but they re-created their git from svn repo at some point). Qt port doesn't exist in upstream WebKit, it exists only here. Qt Company doesn't have any interest in WebKit since Qt 5.5 release, however their infrastructure was integral part of release process since in order then to build binaries for all supported platforms, with additional benefit of ensured compatibility with official Qt binaries (because builds were made in exactly the same build environments).

**Comment by kloczek** [comment]
> @RJVB More likely, on the wrong planet.
> 
> @kloczek Nope, just don't want to deal with people who treat me like crap

Why do you thing that asking humble question is treating someone like crap? 🤔

**Comment by annulen** [comment]
Asking questions is fine, even if not humble. Sending people 403 because they live of certain country is what I was referring to. Says it all about so called "free software".

![2023-07-27-112523_538x181_scrot](https://github.com/qtwebkit/qtwebkit/assets/102544/70a1646e-1a14-41a3-a282-cf2958dcf806)

**Comment by RJVB** [comment]
I do concur with @kloczek: why do you need access to qt.io? I assume you have sufficiently up-to-date Qt sources available locally from (just) before the start of all those "special operations" and as I said, I think the most important thing would be to update the WebKit part. Personally I'd even prefer it if the Qt glue didn't get updated so much it no longer builds against 5.9 ;)
But there are more potential collaborators who could handle the updating of that code so it also builds against newer Qt versions (and leverages new useful features); you seem to be the only one who was sufficiently motivated and/or knowledgeable to work on the WebKit part...

BTW, there's no mirror of the Qt git repos on here ... and what about using a VPN as so many of your countrymen do to access crucial services like Instagram (which AFAIK are blocked from inside)?

**Comment by annulen** [comment]
>why do you need access to qt.io?

I don't. And I don't want to contribute to Qt anymore, or promote its ecosystem in any way.

>I think the most important thing would be to update the WebKit part

There is no "WebKit part". There is WebKit port to Qt, which is the whole project. Updating API layer is tiniest of all work.

> you seem to be the only one who was sufficiently motivated and/or knowledgeable to work on the WebKit part

Not at all. @mnutt is apparently more competent and motivated than me, and [his fork](https://github.com/movableink/webkit/) contains +/- working update to current WebKit with just a few rough edges. Perhaps something could be released from it.

**Comment by mnutt** [comment]
If I was in @annulen's position I wouldn't feel super motivated to contribute to qt projects, either.

I'd say that on the whole, knowledge of WebKit is more important / much harder to come by than knowledge of qt. Or maybe it's just that there are millions of qt projects out there and ~five WebKit ports. And for that I'm very grateful for any assistance @annulen provides.

**Comment by RJVB** [comment]
>If I was in @annulen's position I wouldn't feel super motivated to contribute to qt projects, either.

I probably wouldn't either (but as noted above Qt as a company have very little interest in QtWebkit and I'd say the ecosystem created by Qt as a product has taken a life beyond the company...). At the same time I can understand Qt's decision to block "export" in light of the current geopolitical state of affairs - if it's even their own decision. (Which is not to say that I think it's the best or even a good decision!)

I concur that any contribution @annulen will still be willing to make will be more than welcome.

**Comment by annulen** [comment]
>I concur that any contribution @annulen will still be willing to make will be more than welcome.

Thanks. If I have free time and nothing more important to do I will consider that :)
